rest-api/dao/participant_summary_dao.py

```diff
@@ -150,6 +150,53 @@ def _get_dna_isolates_sql_and_params():
       params
   )
 
+def _get_sample_status_time_sql_and_params():
+  """Gets SQL that to update enrollmentStatusCoreStoredSampleTime field
+  on the participant summary.
+  """
+
+  dns_test_list = config.getSettingList(config.DNA_SAMPLE_TEST_CODES)
+
+  status_time_sql = '%s' % ','.join(["""COALESCE(sample_status_%s_time, '3000-01-01')""" % item
+                                     for item in dns_test_list])
+
+  sub_sql = """
+  (
+    SELECT 
+      MAX(
+        consent_for_electronic_health_records_time,
+        questionnaire_on_the_basics_time,
+        questionnaire_on_lifestyle_time,
+        questionnaire_on_overall_health_time,
+        physical_measurements_finalized_time,
+        CASE WHEN
+            MIN(
+                {status_time_sql}
+                ) = '3001-01-01' THEN NULL
+            ELSE MIN(
+                {status_time_sql}
+                )
+        END
+      )
+    FROM
+      participant_summary ps
+    WHERE
+      ps.enrollment_status = 3
```

Comment:
I'm not sure you need this WHERE condition or the one on line 186; they duplicate the WHERE conditions of the outer SQL.